examples/lock-app/silabs/src/LockManager.cpp

```diff
@@ -743,25 +918,55 @@ bool LockManager::setLockState(chip::EndpointId endpointId, const Nullable<chip:
         return false;
     }
 
-    // Check the PIN code
-    for (const auto & currentCredential : mLockCredentials[to_underlying(CredentialTypeEnum::kPin)])
+    // Check all pin codes associated to all users to see if this pin code exists
+    for (uint32_t userIndex = 1; userIndex <= kMaxUsers; userIndex++)
     {
+        EmberAfPluginDoorLockUserInfo user;
 
-        if (currentCredential.status == DlCredentialStatus::kAvailable)
+        if (!GetUser(endpointId, userIndex, user))
         {
-            continue;
+            ChipLogError(Zcl, "Unable to get the user - internal error [endpointId=%d,userIndex=%lu]", endpointId, userIndex);
+            return false;
         }
-
-        if (currentCredential.credentialData.data_equal(pin.Value()))
+        // Loop through each credential attached to the user
+        for (uint32_t userCredentialIndex = 0; userCredentialIndex < user.credentials.size(); userCredentialIndex++)
```

Comment:
Might be more readable as:
```suggestion
        for (auto & userCredential : user.credentials)
```
and then using `userCredential` everywhere you have `user.credentials.data()[userCredentialIndex]`.